lib/node.js

```diff
@@ -501,7 +544,38 @@ declare module "tls" {
 }
 
 declare module "url" {
-  // TODO
+  declare function parse(
+    urlStr: string,
+    parseQueryString?: boolean,
+    slashesDenoteHost?: boolean
+  ): {
+    protocol: ?string;
+    slashes: ?string;
+    auth: ?string;
+    host: ?string;
+    port: ?string;
+    hostname: ?string;
+    hash: ?string;
+    search: ?string;
+    query: ?any; // null | string | Object
+    pathname: string;
+    path: string;
+    href: string;
+  };
+  declare function format(urlObj: {
+    href?: string;
```

Comment:
Rather than having to type these out twice, you might consider a type alias similar to what we've done for other APIs:

https://github.com/facebook/flow/blob/master/lib/node.js#L65